site/en/tutorials/load_data/images.ipynb

```diff
@@ -0,0 +1,1356 @@
+      "source": [
+        "Wrap up these up in simple functions for later."
+      ]
+    },
+    {
+      "metadata": {
+        "id": "HmUiZJNU73vA",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "def preprocess_image(image):\n",
+        "  image = tf.image.decode_jpeg(image, channels=3)\n",
+        "  image = tf.image.resize_images(image, [192, 192])\n",
+        "  image /= 255.0  # normalize to [0,1] range\n",
+        "\n",
+        "  return image"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "einETrJnO-em",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "def load_and_preprocess_image(path):\n",
+        "  image = tf.read_file(path)\n",
+        "  return preprocess_image(image)"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "3brWQcdtz78y",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "import matplotlib.pyplot as plt\n",
+        "\n",
+        "image_path = all_image_paths[0]\n",
+        "label = all_image_labels[0]\n",
+        "\n",
+        "plt.imshow(load_and_preprocess_image(img_path))\n",
+        "plt.grid(False)\n",
+        "plt.xlabel(caption_image(img_path))\n",
+        "plt.title(label_names[label].title())\n",
+        "print()"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "n2TCr1TQ8pA3",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "## Build a `tf.data.Dataset`"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "6H9Z5Mq63nSH",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "### A dataset of images"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "GN-s04s-6Luq",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "The easiest way to build a `tf.data.Dataset` is using the `from_tensor_slices` method.\n",
+        "\n",
+        "Slicing the array of strings, results in a dataset of strings:"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "6oRPG3Jz3ie_",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "path_ds = tf.data.Dataset.from_tensor_slices(all_image_paths)"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "uML4JeMmIAvO",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "The `output_shapes` and `output_types` fields describe the content of each item in the dataset. In this case it is a set of scalar binary-strings"
```

Comment:
We're going to remove these properties in V2. (They were rarely used by user code, and they aren't sufficient to express all of the things that can be in a dataset... especially the nested output of `Dataset.window()`.) The replacement will most likely be a single `Dataset.element_structure` property (exact name TBD) that serves the same purpose, and provides a superset of the same information.